.github/workflows/rust-test.yml

```diff
@@ -46,3 +46,32 @@ jobs:
         working-directory: ${{ inputs.working-directory }}
         run: |
           cargo test -- --test-threads=1
+
+  test_clang10:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v3
+
+      - name: Setup toolchain
+        uses: dtolnay/rust-toolchain@master
```

Comment:
Is this recommended over `actions-rs/toolchain@*`?

Reply:
Good question!

`actions-rs` stuff has not been updated in 2-3 years and are dumping tons of deprecation warnings into our GHA logs, which we really should get upgraded because GitHub didn't give a timeline on when things would get disabled. In these reusable workflows, I took the opportunity to replace the `actions-rs/toolchain` action with https://github.com/dtolnay/rust-toolchain which seems to be an up-to-date action with similar behavior (and [David Tolnay](https://github.com/dtolnay) is a highly respected Rust dev). This means that we no longer have deprecation warnings in every run.

Reply:
Ok perfect